external_push/models.py

```diff
@@ -471,3 +471,160 @@ def send_data(self):
 
         r = requests.post(self.logging_url, data=json_data, headers=headers)
         return True  # TODO - Check if the post actually succeeded & react accordingly
+
+        
+        
+        
+class ThingSpeakPushTarget(models.Model):
+    class Meta:
+        verbose_name = "ThingSpeak Push Target"
+        verbose_name_plural = "ThingSpeak Push Targets"
+
+    STATUS_ACTIVE = 'active'
+    STATUS_DISABLED = 'disabled'
+    STATUS_ERROR = 'error'
+
+    STATUS_CHOICES = (
+        (STATUS_ACTIVE, 'Active'),
+        (STATUS_DISABLED, 'Disabled'),
+        (STATUS_ERROR, 'Error'),
+    )
+
+    SENSOR_SELECT_ALL = "all"
+    SENSOR_SELECT_LIST = "list"
+    SENSOR_SELECT_NONE = "none"
+
+    SENSOR_SELECT_CHOICES = (
+        (SENSOR_SELECT_ALL, "All Active Sensors/Devices"),
+        (SENSOR_SELECT_LIST, "Specific Sensors/Devices"),
+        (SENSOR_SELECT_NONE, "Nothing of this type"),
+    )
+
+    PUSH_FREQUENCY_CHOICES = (
+        # (30-1,    '30 seconds'),
+        (60 - 1, '1 minute'),
```

Comment:
Do you happen to know if ThingSpeak has a "minimum" push frequency? Is 1 minute too short, or is it allowed?

Reply:
The free tier provides for 3,000,000 pushes per year  (8,219 messages per day) and up to 4 channels sharing that data.  I do not believe there is a push frequency limit published. (Read that I can't find it)

Reply:
8219 would be roughly once every 30 seconds, so I think we’re good at 1/min!

Reply:
The limit works for 1 minute but if going for multiple channels (Brewpis) the user would have to draw back on the update frequency if using all the time.  I did notice that you have to create a beer profile before the push works.  I had a lot of head scratching as I had my fridge on "Fridge Constant" without a beer profile.   Makes scene once I did create one.